extensions/sql-database-projects/src/tools/netcoreTool.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as os from 'os';
+import { isNullOrUndefined } from 'util';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+export const DBProjectConfigurationKey: string = 'sqlDatabaseProjects';
+export const NetCoreInstallLocationKey: string = 'netCoreSDKLocation';
+export const NetCoreInstallationConfirmation: string = localize('sqlDatabaseProjects.NetCoreInstallationConfirmation', "The .NET Core SDK cannnot be located. Project build will not work. Please install the same or update the .Net Core SDK location in settings if already installed.");
+export const UpdateNetCoreLocation: string = localize('sqlDatabaseProjects.UpdateNetCoreLocation', "Update .Net Core location");
+export const InstallNetCore: string = localize('sqlDatabaseProjects.InstallNetCore', "Install .Net Core SDK");
+
+export class NetCoreTool {
+
+	public findOrInstallNetCore(): void {
+		if (!this.isNetCoreInstallationPresent) {
+			this.showInstallDialog();
+		}
+	}
+
+	private showInstallDialog(): void {
+		vscode.window.showInformationMessage(NetCoreInstallationConfirmation, UpdateNetCoreLocation, InstallNetCore).then(async (result) => {
+			if (result === UpdateNetCoreLocation) {
+				//open settings
+				vscode.commands.executeCommand('workbench.action.openGlobalSettings');
+			}
+			else if (result === InstallNetCore) {
+				//open install link
+				const dotnetcoreURL = 'https://dotnet.microsoft.com/download/dotnet-core/3.1';
+				vscode.env.openExternal(vscode.Uri.parse(dotnetcoreURL));
+			}
+		});
+	}
+
+	public get isNetCoreInstallationPresent(): Boolean {
+		return (!isNullOrUndefined(this.netcoreInstallLocation) && fs.existsSync(this.netcoreInstallLocation));
```

Comment:
probably should be defaulting to async for fs operations in case something's being particularly slow on that front

Reply:
good point but we use this everywhere since fs.exists is not symply async but callback based however here I need tocheck from multiple path and return to single pop up. Leaving this for now - will check on perf.